### PR TITLE
style: ban js types in favour of ts types

### PR DESCRIPTION
**Pull request by andersfylling**

### CODE REVIEW

**Comment by andersfylling** [comment]
is Record better than object? 
https://github.com/microsoft/TypeScript/issues/21732

File: .eslintrc.js [hunk 1]
```diff
@@ -29,8 +29,34 @@ module.exports = {
...
+            },
+            "symbol": {
+              "message": "Use ts primitive 'symbol' instead",
+              "fixWith": "symbol"
+            },
+
+            "{}": {
+              "message": "Use object instead",
+              "fixWith": "object"
+            }
```
Comment (rek) [inline .eslintrc.js]: object is not recommended to use anymore, it's too general

File: .eslintrc.js [hunk 2]
```diff
@@ -29,8 +29,34 @@ module.exports = {
...
+            },
+            "Boolean": {
+              "message": "Use ts primitive 'boolean' instead",
+              "fixWith": "boolean"
+            },
+            "Number": {
+              "message": "Use ts primitive 'number' instead",
+              "fixWith": "number"
+            },
+            "symbol": {
```
Comment (rek) [inline .eslintrc.js]: not sure about symbol either. this one is pretty rare

**Comment by rek** [comment]
fyi https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/configs/recommended.ts

**Comment by andersfylling** [comment]
Closing as it's become stale and outdated.

**Pull request closed without merging**